### PR TITLE
Fix crash related to audio on iOS <14.5

### PR DESCRIPTION
**Pull request by AGulev**
Fixes issue when application crashes on start because of `AVFAudion` framework.
Minimum iOS version increased to 9.0, it's a requirement for Xcode 13. `Info.plist` file updated. If you use custom `Info.plist` make sure you apply the same change for `MinimumOSVersion`.

### CODE REVIEW

File: .github/workflows/main-ci.yml [hunk 1]
```diff
@@ -111,7 +111,7 @@ jobs:
     strategy:
       matrix:
         platform: [arm64-darwin, x86_64-ios]
-    runs-on: ubuntu-20.04
+    runs-on: macOS-latest
```
Comment (AGulev) [inline .github/workflows/main-ci.yml:114]: We have issues with frameworks linking if build on ubuntu.
Here is diff in `otool -L ` between Mac and ubuntu:

```
Mac:
	/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation (compatibility version 150.0.0, current version 1856.105.0, weak)

Ubuntu:
	/System/Library/Frameworks/AVFAudio.framework/AVFAudio (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation (compatibility version 150.0.0, current version 1856.105.0)

```

File: build_tools/sdk.py [hunk 2]
```diff
@@ -43,7 +43,7 @@
 
 # NOTE: Minimum iOS-version is also specified in Info.plist-files
 # (MinimumOSVersion and perhaps DTPlatformVersion)
-VERSION_IPHONEOS_MIN="8.0"
+VERSION_IPHONEOS_MIN="9.0"
```
Comment (AGulev) [inline build_tools/sdk.py:46]: minimum iOS sdk for Xcode 13 is iOS 9.0

File: build_tools/waf_dynamo.py [hunk 3]
```diff
@@ -367,7 +367,7 @@ def default_flags(self):
     if 'osx' == build_util.get_target_os() or 'ios' == build_util.get_target_os():
         self.env.append_value('LINKFLAGS', ['-weak_framework', 'Foundation'])
         if 'ios' == build_util.get_target_os():
-            self.env.append_value('LINKFLAGS', ['-framework', 'UIKit', '-framework', 'SystemConfiguration', '-framework', 'AVFAudio'])
+            self.env.append_value('LINKFLAGS', ['-framework', 'UIKit', '-framework', 'SystemConfiguration', '-framework', 'AVFoundation'])
```
Comment (AGulev) [inline build_tools/waf_dynamo.py:370]: Audio functionality split into `AVFAudio` from `AVFoundation` in iOS 14.5
But we should use AVFoundation to keep back compatibility. I'm not sure how it resolved inside SDK, pbbly `AVFAudio`  is linked in  `AVFoundation` or something like that

File: engine/sound/src/sound_ios.mm [hunk 4]
```diff
@@ -165,10 +165,7 @@ - (void) handleRouteChange:(NSNotification *) notification {
 // ---------------------------------------------------------------------------
 namespace dmSound
 {
-    bool AudioSessionInitialized = false;
```
Comment (AGulev) [inline engine/sound/src/sound_ios.mm:168]: this is unused flag.

**Review by JCash — APPROVED** [review]

File: share/extender/build.yml [hunk 5]
```diff
@@ -211,7 +211,7 @@ platforms:
         LD_LIBRARY_PATH:        "{{env.LIB_TAPI_1_6_PATH}}:{{env.LD_LIBRARY_PATH}}"
         SYSROOT:                "{{env.PLATFORMSDK_DIR}}/iPhoneSimulator{{env.IOS_15_VERSION}}.sdk"
     context:
-        frameworks: ["OpenGLES", "OpenAL", "QuartzCore", "CoreGraphics", "AudioToolbox", "SystemConfiguration", "CoreVideo", "UIKit", "Security", "AVFAudio"]
```
Comment (AGulev) [inline share/extender/build.yml:214]: it seems like we don't use `Security` framework at all, so I remove that

**Pull request merged**